### PR TITLE
Fix race on force deleting container created by task

### PR DESCRIPTION
**Pull request by tonistiigi**
If a container gets deleted with `rm -f` it may not exist after `wait` has returned. This moves the inspect call before `wait`.

@dongluochen 

Signed-off-by: Tonis Tiigi tonistiigi@gmail.com

### CODE REVIEW

**Comment by vieux** [comment]
LGTM

**Comment by stevvooe** [comment]
How will this pick up the correct exit code?

File: daemon/cluster/executor/container/controller.go [hunk 1]
```diff
@@ -151,23 +151,21 @@ func (r *controller) Wait(pctx context.Context) error {
 	ctx, cancel := context.WithCancel(pctx)
 	defer cancel()
 
+	ctnr, err := r.adapter.inspect(ctx)
+	if err != nil {
+		// TODO(stevvooe): Need to handle missing container here. It is likely
+		// that a Wait call with a not found error should result in no waiting
+		// and no error at all.
+		return err
+	}
```
Comment (dongluochen) [inline daemon/cluster/executor/container/controller.go]: The check `if ctnr.State.ExitCode != 0 {` should follow this.

**Comment by icecrime** [comment]
@tonistiigi Need a rebase :disappointed:

**Comment by icecrime** [comment]
@tonistiigi

**Comment by mlaventure** [comment]
@tonistiigi 
```
22:23:33 container/state_test.go:22: invalid operation: s.ExitCode != 0 (mismatched types func() int and int)
22:23:33 container/state_test.go:37: invalid operation: s.ExitCode != i (mismatched types func() int and int)
```

**Comment by stevvooe** [comment]
LGTM

**Comment by mlaventure** [comment]
@tonistiigi 

You forgot windows
```
22:39:33 container\state_windows.go:6: cannot assign to s.ExitCode
22:39:33 container\state_windows.go:6: cannot use exitStatus.ExitCode (type int) as type func() int in assignment
```

**Comment by tonistiigi** [comment]
@mlaventure @icecrime PTAL

**Comment by icecrime** [comment]
@tonistiigi Windows doesn't build:

```
16:40:53 # github.com/docker/docker/daemon
16:40:53 daemon\monitor_windows.go:32: cannot assign to container.CommonContainer.State.ExitCode
16:40:53 daemon\monitor_windows.go:32: cannot use -1 (type int) as type func() int in assignment
```

**Comment by tonistiigi** [comment]
```
Building: bundles/1.12.0-dev/cross/windows/amd64/docker-1.12.0-dev.exe
Created binary: bundles/1.12.0-dev/cross/windows/amd64/docker-1.12.0-dev.exe
Building: bundles/1.12.0-dev/cross/windows/amd64/dockerd-1.12.0-dev.exe
Created binary: bundles/1.12.0-dev/cross/windows/amd64/dockerd-1.12.0-dev.exe

# echo $?
0
```

**Pull request merged**